pykafka/simpleconsumer.py

```diff
@@ -525,10 +523,18 @@ def _handle_success(parts):
                     for errcode, parts in parts_by_error.iteritems()
                     for part, _ in parts)
 
-            # release all locks to allow fetching
             for errcode, owned_partitions in parts_by_error.iteritems():
                 if errcode != 0:
                     for owned_partition in owned_partitions:
+                        # set internal counter appropriately to avoid possibly
+                        # leaving it in a state inconsistent with the given
+                        # offset
+                        given_offset = owned_partition_offsets[owned_partition]
+                        # don't reset counter to a magic value, only to a real
+                        # offset
+                        if given_offset != self._auto_offset_reset:
```

Comment:
Not entirely sure if I got that bit, but could `given_offset` also be a timestamp at this point?